### PR TITLE
Issue #1669 - fix composite search queries

### PR DESCRIPTION
**Pull request by michaelwschroeder**
Signed-off-by: Mike Schroeder <mschroed@us.ibm.com>

### CODE REVIEW

**Comment by michaelwschroeder** [comment]
There is already a set of tests for composite search (AbstractSearchCompositeTest), they just weren't enabled in the testng.xml file. So I didn't create any new tests, just enabled the existing tests to be run.

**Review by lmsurpre — APPROVED** [review]
LGTM

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java [hunk 1]
```diff
@@ -826,12 +826,16 @@ private void appendInnerSelect(StringBuilder whereClauseSegment, QueryParameter
                 for (int componentNum = 1; componentNum <= components.size(); componentNum++) {
                     String alias = chainedParmVar + "_p" + componentNum;
                     QueryParameter component = components.get(componentNum - 1);
+                    String tableName = QuerySegmentAggregator.tableName(resourceTypeName, component);
```
Comment (punktilious) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java:829]: Might be worth a comment noting the issue number and the fact that composites still use the old token values table (for now)

**Review by punktilious — APPROVED** [review]
LGTM

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java (hunk 1 above)
Comment (michaelwschroeder) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java:829]: Done.

**Pull request merged**